### PR TITLE
Personal Info page fixes.

### PR DESCRIPTION
**Pull request by harrymaynard-maximus**
### Please check if the PR fulfills these requirements:

- [x] After these changes, the app was run and still works as expected
- [ ] Unit tests for these changes were added (if applicable)
- [x] All existing unit tests were run and still pass

### CODE REVIEW

**Review by bryce-mcmath — APPROVED** [review]
I thought the SIN number was required only for supp ben. Could be wrong. Looks good otherwise

**Comment by harrymaynard-maximus** [comment]
The wireframe shows the SIN field for SB and FPC.

File: jha/src/helpers/validators.js [hunk 1]
```diff
@@ -16,3 +21,29 @@ export const nonBCValidator = (value) => {
...
+  return () => {
+    if (!dateData || !dateData.year || typeof dateData.month !== 'number' || !dateData.day) {
+      return true;
+    }
+    const year = dateData.year;
+    const month = dateData.month;
+    const day = dateData.day;
+    if (!(year && typeof month === 'number' && day)
+      && (year || typeof month === 'number' || day)) {
+      return false;
```
Comment (Ian-c-cameron) [inline jha/src/helpers/validators.js:44]: line 44 will never execute. If year is falsey then it has already returned.  if typeof month isn't 'number' then it has already returned, and if day is falsey then it has already returned, so in line 42/43 the if statement reads if( false && true) which translates to if(false) which will never execute.

File: jha/src/helpers/validators.js [hunk 2]
```diff
@@ -16,3 +21,29 @@ export const nonBCValidator = (value) => {
...
+    }
+    const isoDateString = getISODateString(year, month + 1, day);
+    return isValidISODateString(isoDateString);
```
Comment (Ian-c-cameron) [inline jha/src/helpers/validators.js:47]: isValidISODateString will always be true when testing the result of getISODateString unless year given to it has more than 4 digits, the day has more than 2 digits, or the month has more than two digits.  Being that the data being handed to it is originating in a DateInput Component, it is not possible for this to be the case.  Therefore dateDataValidator can only ever return true(see previous comment).

**Review by Ian-c-cameron — CHANGES_REQUESTED** [review]
I am a bit confused about the use of a processed date for validation as opposed to validating the JS Date object retuned to the v-model.

**Comment by harrymaynard-maximus** [comment]
The `processDate` event of the `DateInput` component is emitted whenever the user inputs data (on blur). Since the user could theoretically enter a day of "32", the `processDate` event is emitted and sends the data as the user has entered it. In contrast, the `v-model` value is only set when a valid `Date` object instance is created, which only happens when the input values can create a valid date.

There's currently no way to differentiate between an empty set of fields and an invalid date in `v-model` since the value for both would be `null`. Since we have to have separate error messages for "Date is required" and "Date is invalid", that is why I have to use the data (inputted values) from the `processDate` event to validate the date.

This definitely isn't the most ideal solution. It would be nice to have a specific value sent to `v-model` when a date is given but not valid. The `DateInput` component is already used by 3 projects now, so refactoring would impact them unfortunately.

File: jha/src/helpers/validators.js (hunk 2 above)
Comment (harrymaynard-maximus) [reply jha/src/helpers/validators.js:47]: `getISODateString` could theoretically return "YYYY-MM-32" if it's supplied those values. That's why the `isValidISODateString` is needed.

**Comment by Ian-c-cameron** [comment]
Ok... that explains it... it may be a good idea to emit 'invalid' event with a boolean value instead of a processed date if the date is invalid... leave the processDate emitter in place so that we don't have to imediately change other projects.  This would negate the need for a complex validator for the processDate output.

**Pull request merged**